.github/workflows/daily-build-renew-cache.yml

```diff
@@ -0,0 +1,36 @@
+name: Daily Build Renew Cache
+on:
+  schedule:
+    - cron: '0 7 * * *'
+
+jobs:
+  docker:
+    name: Build Docker with Cache
+    runs-on: ubuntu-20.04
+    environment: production
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Determine Meta
+        run: echo "REF=${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+      - name: Dockerhub Login
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Docker Build and Push (renew cache)
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          cache-to: type=registry,ref=${{ secrets.DOCKER_REPOSITORY }}:${{ env.REF }}-cache,mode=max
+          push: true
+          build-args: |
+            GIT_REPO=https://github.com/${{ github.repository }}.git
+            GIT_REF=${{ env.REF }}
+          tags: ${{ secrets.DOCKER_REPOSITORY }}:${{ env.REF }}
```

Comment:
Is there some particular reason to store the docker repository in a repo secret?

Reply:
This is the only method I'm aware of to pass in variables.   This makes the repo portable so when someone forks it, they can set the environment variable and the github action doesn't need to be changed.